### ISSUE TITLE
Is this a typo in recommenders/docs/examples/basic_retrieval.ipynb?

### ISSUE THREAD

**Issue by poinwater** [tensorflow/recommenders#299]
In class ModelensModel and NoBaseClassMovielensModel, the line of self.task in the __init__ is:

`self.task: tf.keras.layers.Layer = task `

But in the quickstart.ipynb, the self.task in class Modelens is:

`task: tfrs.tasks.Retrieval`

I'm wondering if the task class belongs to the keras.layers.Layer class. Is this a typo?

**Comment by poinwater** [comment]
Thank you for your reply. 

I'm writing an essay about the recommendation system, but I got confused by the basics tutorial. Could you answer a few questions? I really appreciate it.

In the Recommending basics, the notebook says "the real-world recommender systems are often composed of two stages, retrieval stage, and ranking stage. And the ranking stage takes the outputs of the retrieval model and fin-tunes them. "

But in the notebooks, the retrieval model and ranking model are almost independent. The only sharing part between the two models is the embeddings layers, but the layers have initialized independently. The full model in the ranking notebooks seems just the ranking model with the Ranking task(the combination of loss and metrics). The ranking model doesn't take the outputs of the retrieval model but directly takes the models and rating information as inputs.  

So here is my first question: Is the full model part just the compiled retrieval model or ranking model rather than the combination of retrieval model and ranking model?

My second question is about prediction. In the ranking notebook, the following code allows the ranking model to predict a sample: 
`RankingModel()((["42"], ["One Flew Over the Cuckoo's Nest (1975)"]))`

But the score is different every time I run it. And the Ranking model predicts the sample before it has been trained, shall we use the trained full model to predict the sample?

Here are the prediction outputs, 
Running the first time:
![image](https://user-images.githubusercontent.com/56426474/119207482-3473fa80-ba53-11eb-9f34-643b610ee4f0.png)
Running the second time:
![image](https://user-images.githubusercontent.com/56426474/119207488-405fbc80-ba53-11eb-9f93-d3430b6ec43a.png)

**Comment by poinwater** [comment]
Besides, the tutorial says the code can predict the recommendations for user 42.
But I replaced the number with 99999999, and get a result without error.

Does this code really predict recommendations according to user_id?
![image](https://user-images.githubusercontent.com/56426474/119280133-ce1ee180-bbe4-11eb-9f70-45e488b96734.png)

**Comment by maciejkula** [comment]
In your example you're creating the model from scratch on the second call. This gives you new, completely random weights. That's why you get different predictions.
